pkg/cmd/experimental/diagnostics/diagnostics.go

```diff
@@ -163,21 +168,28 @@ only these are available:
 The list of all possible is:
     {{.available}}
 		`, log.Hash{"requested": o.RequestedDiagnostics, "common": common.List(), "available": available.List()}))
-	}
 
-	// If not given master/client config file locations, check if the defaults exist
-	// and adjust the options accordingly:
-	if len(o.MasterConfigLocation) == 0 {
-		if _, err := os.Stat(StandardMasterConfigPath); !os.IsNotExist(err) {
-			o.MasterConfigLocation = StandardMasterConfigPath
-		}
-	}
-	if len(o.NodeConfigLocation) == 0 {
-		if _, err := os.Stat(StandardNodeConfigPath); !os.IsNotExist(err) {
-			o.NodeConfigLocation = StandardNodeConfigPath
-		}
+		return fmt.Errorf("Not all requested diagnostics are available: requested=%s available=%s", strings.Join(o.RequestedDiagnostics, ", "), strings.Join(available.List(), ", "))
 	}
 
+	return nil
+}
+
+func availableDiagnostics() sets.String {
+	available := sets.NewString()
+	available.Insert(availableClientDiagnostics.List()...)
+	available.Insert(availableClusterDiagnostics.List()...)
+	available.Insert(availableHostDiagnostics.List()...)
+	return available
+}
+
+// RunDiagnostics builds diagnostics based on the options and executes them, returning a summary.
+func (o DiagnosticsOptions) RunDiagnostics() (bool, error, int, int) {
```

Comment:
@sosiouxme not a big fan of `bool, error, int, int` - we have objects capable of expressing "this is a collection of errors, some of which are warnings and some failures". They could potentially be used here to simplify things a bit.